apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -145,8 +154,13 @@ private void publishEvent(
     throws MetaException {
     JSONObject json = createBaseMessage(eventType, table.getDbName(), table.getTableName());
 
+    json.put("tableLocation", table.getSd().getLocation());
+
+    json.put("tableParameters", getFilteredParams(table.getParameters()));
```

Comment:
Hmmm, I can see why having empty params in the message might be useful if you provided a filter but nothing matched. This would be different to not providing a filter at all in which case I think the params element shouldn't end up in the message. Anyone else have an opinion? Easy to implement either way, just need some logic, a test and some documentation depending what we decide.

Reply:
Yeah. I think having an empty map makes all events more consistent with different scenarios and the receiver of these messages wouldn't need to handle different scenarios at their end. They can always just assume that there is a map with zero or more params that will come.

Reply:
For a scenario where REGEX doesn't match any param in table parameters, an empty map will be returned.